### PR TITLE
Qos interfaces

### PR DESCRIPTION
**Pull request by ross-desmond**
*Description of changes:*
Adds QoS interfaces with no-op and return unsupported for fastrtps.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by dabonnie — COMMENTED** [review]
We need to rebase and pull changes from upstream.

File: rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/custom_subscriber_info.hpp [hunk 1]
```diff
@@ -102,15 +102,15 @@ class SubListener : public eprosima::fastrtps::SubscriberListener
   }
 
   void
-  data_taken()
+  data_taken(eprosima::fastrtps::Subscriber * sub)
```
Comment (dabonnie) [inline rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/custom_subscriber_info.hpp]: Please add docstring for the new input param.

**Comment by mm318** [comment]
> We need to rebase and pull changes from upstream.

I merged in changes from upstream. The diff should be more meaningful now.

File: rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/rmw_common.hpp [hunk 2]
```diff
@@ -1,313 +1,313 @@
...
-__rmw_create_wait_set(const char * identifier, rmw_context_t * context, size_t max_conditions);
-
-RMW_FASTRTPS_SHARED_CPP_PUBLIC
-rmw_ret_t
-__rmw_destroy_wait_set(const char * identifier, rmw_wait_set_t * wait_set);
-
-}  // namespace rmw_fastrtps_shared_cpp
-
-#endif  // RMW_FASTRTPS_SHARED_CPP__RMW_COMMON_HPP_
+// Copyright 2016-2018 Proyectos y Sistemas de Mantenimiento SL (eProsima).
```
Comment (mm318) [inline rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/rmw_common.hpp:1]: Let's keep this change. `rmw_common.hpp` was previously Windows line endings and we'll be changing it to Unix line endings.

File: rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/custom_subscriber_info.hpp (hunk 1 above)
Comment (mm318) [reply rmw_fastrtps_shared_cpp/include/rmw_fastrtps_shared_cpp/custom_subscriber_info.hpp]: I think this change came from ros2/master.

**Review by mm318 — APPROVED** [review]

**Pull request merged**